lib/Controller/MappingsController.php

```diff
@@ -253,10 +271,15 @@ public function test(): JSONResponse
         $validationErrors = [];
 
         // Perform schema validation if both schema and validation are provided
-        if ($schema !== false && $validation !== false) {
-            // TODO: Implement schema validation logic here
-            // For now, we'll just assume it's always valid
-            $isValid = true;
+        if ($schema !== false && $validation !== false && $openRegisters !== null) {
+			$result = $openRegisters->validateObject(object: $resultObject, schemaObject: $schema->getSchemaObject($urlGenerator));
+
+			$isValid = $result->isValid();
+
+			if($result->hasError() === true) {
+				// Class imported without use because it only exists when OpenRegisters is installed.
+				$validationErrors = (new \Opis\JsonSchema\Errors\ErrorFormatter())->format(error: $result->error());
```

Comment:
Can we reach this code if openRegisters is not installed? Resulting in an exception?